Microsoft.Toolkit.Uwp.UI.Controls/AdMediator/UwpAdMediator.xaml.cs

```diff
@@ -0,0 +1,377 @@
+            set { SetValue(AdRefreshSecondsProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for AdRefreshSeconds.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty AdRefreshSecondsProperty =
+            DependencyProperty.Register("AdRefreshSeconds", typeof(int), typeof(int), new PropertyMetadata(35));
+
+        public int MaxErrorsPerRefresh
+        {
+            get { return (int)GetValue(MaxErrorsPerRefreshProperty); }
+            set { SetValue(MaxErrorsPerRefreshProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for MaxErrorsPerRefresh.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty MaxErrorsPerRefreshProperty =
+            DependencyProperty.Register("MaxErrorsPerRefresh", typeof(int), typeof(UwpAdMediator), new PropertyMetadata(3));
+
+        // Dispatch timer to fire at each ad refresh interval.
+        private DispatcherTimer myAdRefreshTimer = new DispatcherTimer();
+
+        // Global variables used for mediation decisions.
+        private Random randomGenerator = new Random();
+        private int errorCountCurrentRefresh = 0;  // Prevents infinite redirects.
+        private int adDuplexWeight = 0;            // Will be set by GetAdDuplexWeight().
+
+        // Microsoft and AdDuplex controls for banner ads.
+        private AdControl myMicrosoftBanner = null;
+        private AdDuplex.AdControl myAdDuplexBanner = null;
+
+        private string MicrosoftAppId => (AnalyticsInfo.VersionInfo.DeviceFamily == "Windows.Mobile") ? this.MobileApplicationId : this.DesktopApplicationId;
+
+        private string MicrosoftAdUnit => (AnalyticsInfo.VersionInfo.DeviceFamily == "Windows.Mobile") ? this.MobileAdUnit : this.DesktopAdUnit;
+
+        private int AdWidth => (AnalyticsInfo.VersionInfo.DeviceFamily == "Windows.Mobile") ? this.MobileAdWidth : this.DesktopAdWidth;
+
+        private int AdHeight => (AnalyticsInfo.VersionInfo.DeviceFamily == "Windows.Mobile") ? this.MobileAdHeight : this.DesktopAdHeight;
+
+        public UwpAdMediator()
+        {
+            this.InitializeComponent();
+
+            this.Loaded += UwpAdMediator_Loaded;
+        }
+
+        private void UwpAdMediator_Loaded(object sender, RoutedEventArgs e)
+        {
+            myAdGrid.Width = this.AdWidth;
+            myAdGrid.Height = this.AdHeight;
+
+            adDuplexWeight = GetAdDuplexWeight();
+            RefreshBanner();
+
+            // Start the timer to refresh the banner at the desired interval.
+            myAdRefreshTimer.Interval = new TimeSpan(0, 0, this.AdRefreshSeconds);
+            myAdRefreshTimer.Tick += MyAdRefreshTimer_Tick;
+            myAdRefreshTimer.Start();
+        }
+
+        private int GetAdDuplexWeight()
+        {
+            return 0;
+
+            //// TODO: Change this logic to fit your needs.
```

Comment:
Should this method do something? There's no way for the consuming app to change any of this. This means below when it checks the weight, it's always 0 and will never show the AdDuplex control.